src/lib/prov/tpm2/tpm2_ecc/tpm2_ecc.cpp

```diff
@@ -0,0 +1,295 @@
+      // session.
+      .objectAttributes = ObjectAttributes::render({
+         .fixed_tpm = true,
+         .fixed_parent = true,
+         .sensitive_data_origin = true,
+         .user_with_auth = true,
+         .decrypt = true,  // TODO: Shall we set this?
+         .sign_encrypt = true,
+      }),
+
+      // We currently do not support policy-based authorization
+      .authPolicy = init_empty<TPM2B_DIGEST>(),
+      .parameters =
+         {
+            .eccDetail =
+               {
+                  // Structures Document (Part 2), Section 12.2.3.5
+                  //   If the key is not a restricted decryption key, this field
+                  //   shall be set to TPM_ALG_NULL.
+                  //
+                  // TODO: Once we stop supporting TSS < 4.0, we could use
+                  //       `.keyBits = {.null = {}}, .mode = {.null = {}}`
+                  //       which better reflects our intention here.
+                  .symmetric =
+                     {
+                        .algorithm = TPM2_ALG_NULL,
+                        .keyBits = {.sym = 0},
+                        .mode = {.sym = TPM2_ALG_NULL},
+                     },
+
+                  // Structures Document (Part 2), Section 12.2.3.6
+                  //   If the decrypt attribute of the key is SET, then this shall be a
+                  // valid key exchange scheme or TPM_ALG_NULL
+                  //
+                  // TODO: Once we stop supporting TSS < 4.0, we could use
+                  //       `.details = {.null = {}}`
+                  //       which better reflects our intention here.
+                  .scheme =
+                     {
+                        .scheme = TPM2_ALG_NULL,
+                        .details = {.anySig = {.hashAlg = TPM2_ALG_NULL}},
+                     },
+                  .curveID = curve_id.value(),
+
+                  // Structures Document (Part 2), Section 12.2.3.6
+                  // If the kdf parameter associated with curveID is not
+                  // TPM_ALG_NULL then this is required to be NULL.
+                  // NOTE There are currently no commands where this parameter
+                  // has effect and, in the reference code, this field needs to
+                  // be set to TPM_ALG_NULL
+                  // TODO: Easier initialization?
+                  .kdf = {.scheme = TPM2_ALG_NULL, .details = {.kdf2 = {.hashAlg = TPM2_ALG_NULL}}},
+               },
+         },
+
+      // For creating an asymmetric key this value is not used.
+      .unique = {.ecc = {}},
+   };
+
+   return create_transient_from_template(
+      ctx, sessions, parent.handles().transient_handle(), key_template, sensitive_data);
+}
+
+namespace {
+
+SignatureAlgorithmSelection make_signature_scheme(std::string_view hash_name) {
+   return {
+      .signature_scheme =
+         TPMT_SIG_SCHEME{
+            .scheme = TPM2_ALG_ECDSA,  // Only support ECDSA
```

Comment:
We might have to explicitly support `TPM2_ALG_SM2` here, when we're dealing with a SM2 keys.

Reply:
True, actually supporting SM2 still requires some changes but I'd say the scope here is only ECDSA